flink-clients/src/main/java/org/apache/flink/client/program/Client.java

```diff
@@ -100,6 +102,9 @@
 	 */
 	private JobID lastJobID;
 
+	/** Switch for blocking/detached job submission of the client */
+	private boolean detachedJobSubmission = false;
```

Comment:
There is a flag to toggle detached submission and there are detached and blocking methods